### ISSUE TITLE
I don't know what's wrong with https://www.google.com + geventhttpclient

### ISSUE THREAD

**Issue by duy13** [geventhttpclient/geventhttpclient#157]
GOOGLE.COM PROBLEM:

```
>>> url = URL('https://www.google.com')



>>> http = HTTPClient.from_url(url)
>>> response = http.get(url.request_uri)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/xxx/python/lib/python2.7/site-packages/geventhttpclient/client.py", line 266, in get
    return self.request(METHOD_GET, request_uri, headers=headers)
  File "/xxx/python/lib/python2.7/site-packages/geventhttpclient/client.py", line 226, in request
    sock = self._connection_pool.get_socket()
  File "/xxx/python/lib/python2.7/site-packages/geventhttpclient/connectionpool.py", line 166, in get_socket
    return self._create_socket()
  File "/xxx/python/lib/python2.7/site-packages/geventhttpclient/connectionpool.py", line 127, in _create_socket
    raise first_error
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
>>> print(response)
<HTTPSocketPoolResponse status=301 headers={'content-length': '193', 'content-type': 'text/html; charset=utf-8', 'x-cache': 'CONFIG_NOCACHE', 'set-cookie': '_EDGE_V=1; domain=bing.com; expires=Thu, 11-May-2023 05:38:13 GMT; path=/; HttpOnly', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'location': 'https://www.bing.com:443/?toWww=1&redig=70DCA1D4865E42329FDEFD50E002A874', 'cache-control': 'private', 'date': 'Sat, 16 Apr 2022 05:38:13 GMT', 'x-msedge-ref': 'Ref A: ED41201533384CC3A2E3ED7F3A68103C Ref B: SIN30EDGE0416 Ref C: 2022-04-16T05:38:13Z', 'x-snr-routing': '1', 'accept-ch': 'Sec-CH-UA-Arch, Sec-CH-UA-Bitness, Sec-CH-UA-Full-Version, Sec-CH-UA-Mobile, Sec-CH-UA-Model, Sec-CH-UA-Platform, Sec-CH-UA-Platform-Version'}>
>>>
```

BUT BING.COM HAS NO ERROR:

```
>>> url = URL('https://www.bing.com')
>>> http = HTTPClient.from_url(url)
>>> response = http.get(url.request_uri)
>>> print(response)
<HTTPSocketPoolResponse status=302 headers={'content-length': '149', 'expires': '-1', 'content-type': 'text/html; charset=utf-8', 'x-cache': 'CONFIG_NOCACHE', 'set-cookie': '_SS=SID=0F5A6843D25A6191088379CBD35B603F; domain=.bing.com; path=/', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'location': 'https://www2.bing.com/?form=DCDN', 'pragma': 'no-cache', 'cache-control': 'no-cache', 'date': 'Sat, 16 Apr 2022 05:44:28 GMT', 'p3p': 'CP="NON UNI COM NAV STA LOC CURa DEVa PSAa PSDa OUR IND"', 'x-snr-routing': '1', 'x-msedge-ref': 'Ref A: B24DB9803ABE4819BBE8FCEA0B4B8929 Ref B: SIN30EDGE0320 Ref C: 2022-04-16T05:44:29Z', 'accept-ch': 'Sec-CH-UA-Arch, Sec-CH-UA-Bitness, Sec-CH-UA-Full-Version, Sec-CH-UA-Mobile, Sec-CH-UA-Model, Sec-CH-UA-Platform, Sec-CH-UA-Platform-Version'}>
```

CENTOS 7 AND DEBIAN 9 HAS NO THAT ERROR, BUT CENTOS 8, DEBIAN 10/11, UBUNTU 18/20 HAS THAT ERROR

`ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)`

I DON'T UNDERSTAND WHAT HAPPENED HERE, I TRYED BOTH PYTHON 2 AND 3 SAME PROBLEM

THANKS FOR ANY SUGGESTIONS!

**Comment by djpnewton** [comment]
I am seeing a similar issue on google websites. From a test of some popular domains:

```
https://www.youtube.com
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
https://www.google.co.in
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
https://www.google.de
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
https://www.google.co.uk
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
https://www.google.fr
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
https://www.google.com.br
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
```

**Comment by cyberw** [comment]
Could this be the underlying issue? https://github.com/certifi/python-certifi/issues/197

Maybe test installing the previous version of certifi?

**Comment by cyberw** [comment]
Perhaps the issue is with `HTTPClient.from_url`? I got the same error with your code, but the following works:

```
from geventhttpclient import URL, HTTPClient
url = URL('https://www.google.com')
http = HTTPClient(url.host)
response = http.get(url.request_uri)
```

**Comment by cyberw** [comment]
I have the same issue myself (on macOS Monterey), but I was unable to fix it. I think it is something to do with certifi using/not using the system certificates.

**Comment by duy13** [comment]
This is my way:
`http = HTTPClient.from_url(url, insecure=True, ssl_context_factory=insecure_ssl_context_factory)`

**Comment by cyberw** [comment]
I get the same issue with 1.x and 2.0, but I dont have time to investigate :-/ 

Using insecure is useful (and that is what we do in Locust by default), but "someone" should really make a proper fix.

**Comment by cyberw** [comment]
Had to revert the fix as it was a breaking change for Locust.

**Comment by cyberw** [comment]
Fixed again in 2.0.8
